extensions/message-tags.md

```diff
@@ -1,97 +1,143 @@
 
-The final version of the specification will use an unprefixed capability name.
+## Format
 
-## Introduction
+The message pseudo-BNF, as defined in [RFC 1459, section 2.3.1][rfc1459] is extended as follows:
 
-This specification extends the [IRCv3.2 Message Tags](./message-tags-3.2.html) specification in a number of ways:
+    <message>       ::= ['@' <tags> <SPACE>] [':' <prefix> <SPACE> ] <command> <params> <crlf>
+    <tags>          ::= <tag> [';' <tag>]*
+    <tag>           ::= <key> ['=' <escaped value>]
+    <key>           ::= [ '+' ] [ <vendor> '/' ] <sequence of letters, digits, hyphens ('-')>
+    <escaped value> ::= <sequence of any characters except NUL, CR, LF, semicolon (`;`) and SPACE>
+    <vendor>        ::= <host>
 
-* Adds a new capability to indicate base support for message tags
-* Defines a prefix for expressing client-only tags
-* Defines a new command for tag-only messages
-* Increases the size limit for tags
+Not all tag keys have a value.
 
-## Motivation
+The ordering of tags is not meaningful.
 
-Previously, clients were required to negotiate a capability with servers for each
-supported tag. This made tags inappropriate for client-only features. By adding a
-new base capability, this specification allows clients to indicate support for
-receiving any well-formed tag, whether or not it is recognised or used. This also
-frees servers from having to filter individual message tags for each client
-response.
+Individual tag keys MUST only be used a maximum of once per message. Clients
+receiving messages with more than one occurrence of a tag key SHOULD discard all
+but the final occurrence.
 
-The client-only tag prefix allows servers to safely relay untrusted client tags,
-keeping them distinct from server-initiated tags that carry verified meaning.
+## Escaping values
+
+The mapping between characters in tag values and their representation in `<escaped value>` is defined as follows:
+
+| Character       | Sequence in `<escaped value>` |
+|-----------------|-------------------------------|
+| `;` (semicolon) | `\:` (backslash and colon)    |
+| `SPACE`         | `\s`                          |
+| `\`             | `\\`                          |
+| `CR`            | `\r`                          |
+| `LF`            | `\n`                          |
+| all others      | the character itself          |
+
+This escape format is more space efficient than URL-escaping, and ensures that message parts can easily by split on spaces and semi-colons before further parsing.
+
+If a lone `\` exists at the end of an escaped value (with no escape character following it), then there
+SHOULD be no output character. For example, the escaped value `test\` should unescape to `test`.
+
+## Size limit
+
+The size limit for message tags is 4607 bytes, including the leading `'@'` (0x40) and trailing space `' '` (0x20) characters. The size limit for the rest of the message remains unchanged at 512 bytes.
```

Comment:
If this is supposed to act as it does right now, i.e. if clients enable the `message-tags` cap then the limit gets boosted to `4607`, otherwise it stays at the standard `512`, then it should be explicitly mentioned that's how it works in the spec.

Reply:
and/or if we are changing behaviour explicitly, this should be listed in the errata of the spec